### PR TITLE
Added cache functionality

### PR DESCRIPTION
**Pull request by pjol**
[Ticket](https://trello.com/c/88WyeHZe)

Added cacheing functionality to improve response time for repeated requests.

### CODE REVIEW

File: global-bundle.pem [hunk 1]
```diff
@@ -0,0 +1,2952 @@
+-----BEGIN CERTIFICATE-----
```
Comment (justcord10) [inline global-bundle.pem]: you dont want this in here

File: router.js [hunk 2]
```diff
@@ -1,96 +1,139 @@
...
-     res.status(500).send('Error fetching product ' + req.params.id);
-  })
+
+
+  const cached = idCache.get(req.params.id);
+
+  if(cached !== null) {
+    res.send(cached);
+  } else {
+    console.log(req.params, req.params.id);
```
Comment (LukeLarson2) [inline router.js]: please remove console.log here

File: db/ETL.js [hunk 3]
```diff
@@ -19,6 +20,8 @@ const stylesModel = mongoose.model('styles', stylesSchema);
 
 
 const populateData = async (model, path, constructor, page) => {
+  console.log(url);
```
Comment (LukeLarson2) [inline db/ETL.js:23]: remove console.log?

File: server.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ const port = 3000;
 const url = process.env.MONGO_URL;
 const certificate = path.join(__dirname, '/db/mongoose-certificate.pem');
 
+console.log(url);
```
Comment (LukeLarson2) [inline server.js]: remove console.log?

File: .gitignore [hunk 5]
```diff
@@ -130,6 +130,6 @@ dist
 .pnp.*
 
 
-/db/mongoose-certificate.pem
+*.pem
```
Comment (justcord10) [inline .gitignore:133]: good good no certs on github

**Review by LukeLarson2 — APPROVED** [review]
Thanks for making corrections, looks good, cant wait to see performance after load balancing

File: db/k6.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ export const options = {
 }
 
 export default function() {
-  const product = Math.floor(Math.random() * 1000011);
-  http.get(`http://localhost:3000/products/${product}`);
+  const product = Math.floor(Math.random() * 1000);
+  http.get(`http://54.202.83.82:3000/products/1234`);
```
Comment (justcord10) [inline db/k6.js:14]: this should use .env files so you dont have to recommit each time you want to change what you are testing

File: router.js [hunk 7]
```diff
@@ -1,96 +1,139 @@
...
+        relatedCache.put(req.params.id, data.related);
+        res.send(data.related);
+      }
+    })
+    .catch(() => {
+      res.status(500).send('Error fetching related for product ' + req,params.id);
+    })
+  }
+
 })
```
Comment (justcord10) [inline router.js:137]: caching set up for each route looks good

File: server.js [hunk 8]
```diff
@@ -11,6 +11,7 @@ const port = 3000;
 const url = process.env.MONGO_URL;
 const certificate = path.join(__dirname, '/db/mongoose-certificate.pem');
 
+(url);
```
Comment (justcord10) [inline server.js:14]: what does this do?

File: db/k6.js (hunk 6 above)
Comment (justcord10) [reply db/k6.js:14]: or maybe not, since this doesnt affect the codebase, just when you are testing

**Review by justcord10 — APPROVED** [review]
I would say looks good overall, there is a random (url); on line 14 of your server.js file. and left a couple of comments for throughout your code.

**Pull request merged**